hooks/pre/35-change-evr-extension-ownership.rb

```diff
@@ -0,0 +1,25 @@
+# In Katello 4.16, the 'evr' extension is removed from PostgreSQL and integrated into the Katello database via a migration.
+# This hook ensures the 'evr' extension's ownership is transferred to the 'foreman' user so migrations can act on it.
+
+if (local_foreman_db? || devel_scenario?) && execute("rpm -q postgresql-evr", false, false)
+  if app_value(:noop)
+    logger.debug("Would start postgresql service")
+  else
+    start_services(['postgresql'])
+  end
+
+  database = param_value('foreman', 'db_database') || 'foreman'
+  username = param_value('foreman', 'db_username') || 'foreman'
+
+  if local_db_exists?(database)
+    sql = "UPDATE pg_extension SET extowner = (SELECT oid FROM pg_authid WHERE rolname='#{username}') WHERE extname='evr'"
+    if app_value(:noop)
+      logger.debug("Would execute: #{sql} to update ownership of the evr extension")
+    else
+      logger.debug("Updating ownership of the evr extension")
+      execute_preformatted_as('postgres', "psql -d #{database} -c \"#{sql}\"", false, true)
+    end
+  else
+    logger.debug("The foreman database does not exist.")
```

Comment:
Perhaps log the explicit DB name too?
```suggestion
    logger.debug("The Foreman database '#{database}' does not exist.")
```